tasks/modules/Makefile

```diff
@@ -65,26 +65,35 @@ define list_examples
 endef
 
 define aws_provider
-provider \"aws\" {\n  region  = \"$(1)\"\n  profile = \"$(2)\"\n}\n\nprovider \"aws\" {\n  alias   = \"global\"\n  region  = \"us-east-1\"\n  profile = \"$(2)\"\n}\n
+provider \"aws\" {\n  region  = \"$(AWS_REGION)\"\n  profile = \"$(AWS_PROFILE)\"\n}\n\nprovider \"aws\" {\n  alias   = \"global\"\n  region  = \"us-east-1\"\n  profile = \"$(AWS_PROFILE)\"\n}\n
 endef
 
 define azurerm_provider
 provider \"azurerm\" {\n  skip_provider_registration = true\n  features {\n    resource_group {\n      prevent_deletion_if_contains_resources = false\n    }\n  }\n}\n
 endef
 
-define azureado_provider
+define azapi_provider
+provider \"azapi\" {\n  use_cli = true\n  use_msi = false\n}\n
+endef
+
+define azuredevops_provider
 provider \"azuredevops\" {}\n
 endef
 
 define provider_file_path
 $(1)/provider.tf
 endef
 
+define add_provider_details
+	$(if $(findstring hashicorp/aws,$(2)),grep -q "aws" $(1) || bash -c 'echo -e "$(call aws_provider)"' >> $(1),)
+	$(if $(findstring azure/azapi,$(2)),grep -q "azapi" $(1) || bash -c 'echo -e "$(call azapi_provider)"' >> $(1),)
+	$(if $(findstring microsoft/azuredevops,$(2)),grep -q "azuredevops" $(1) || bash -c 'echo -e "$(call azuredevops_provider)"' >> $(1),)
+	$(if $(findstring hashicorp/azurerm,$(2)),grep -q "azurerm" $(1) || bash -c 'echo -e "$(call azurerm_provider)"' >> $(1),)
```

Comment:
could we add provider `random`

Reply:
and one found here https://registry.terraform.io/providers/hashicorp/tls/latest/docs

Reply:
What settings do we need to apply for the random provider? I've never seen it explicitly added to our provider.tf files since it's never prompted me.

Reply:
I'll open a followup PR if this is needed (just let me know via Slack).

Keep in mind that we don't need to provide explicit configuration for every provider in this file like when we declare the providers in the `versions.tf` file; this is only items that cannot be set with environment variables and are required for correct operation of a module.